llvm/test/CodeGen/AMDGPU/GlobalISel/no-ctlz-from-umul-to-lshr-in-postlegalizer.ll

```diff
@@ -0,0 +1,98 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
```

Comment:
There are two interesting things in this test case:

1. It is not reproducible w/o `-O0` argument
2. The `udiv` is actually a UB, and replacing either of its operand will make the crash go away (or the pattern not recognized such that the issue will not happen).

Reply:
You would have an easier time reproducing this by directly emitting a multiply + shift by half bitwidth. it just happens one appears inside the div expansion

Reply:
like this? it doesn't seem to be able to reproduce the issue.

```
define void @test(ptr %p, i32 %a, i32 %b) {
  %mul = mul i32 %a, %b
  %lshr = lshr i32 %mul, 16
  store i32 %lshr, ptr %p, align 4
  ret void
}
```

Reply:
You probably want 64-bit multiply, but I don't see the general combine to form umulh from mul + shift